### DISCUSSION TITLE
1.9.6 Development Environment

### DISCUSSION THREAD

**Discussion by wizgod** [bpatrik/pigallery2#697] [General]
Hi Patrik,

I'm running into some errors trying to run npm install from source 1.9.6. My 1.9.0 source is working fine.

I am curious, what development environment and version of node.js are you using to develop 1.9.6?

Thanks,

Wg

**Comment by bpatrik** [comment]
Package.json should contain the supported node version.

There is a dev doc on what to run if you want to do development.

The release packages contain an already built frontend while doing it from
the main branch you will need to build them manually

-- Sorry for being brief, sent from my phone.

**Comment by wizgod** [comment]
Thanks Patrik, I upgraded node.js to 18.*.

I received the following warnings during install; is this expected?

`npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated formidable@1.2.6: Please upgrade to latest, formidable@v2 or formidable@v3! Check these notes: https://bit.ly/2ZEqIau
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated superagent@3.8.3: Please upgrade to v7.0.2+ of superagent.  We have fixed numerous issues with streams, form-data, attach(), filesystem errors not bubbling up (ENOENT on attach()), and all tests are now passing.  See the releases tab for more information at <https://github.com/visionmedia/superagent/releases>.
npm WARN deprecated tslint@6.1.3: TSLint has been deprecated in favor of ESLint. Please see https://github.com/palantir/tslint/issues/4534 for more information.

added 1898 packages, and audited 1899 packages in 31s

177 packages are looking for funding
  run `npm fund` for details

30 vulnerabilities (13 moderate, 12 high, 5 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.

Process finished with exit code 0
`

**Comment by bpatrik** [comment]
Yes. That is all good

-- Sorry for being brief, sent from my phone.
